libmozevent/utils.py

```diff
@@ -139,10 +140,14 @@ class AsyncRedis(object):
     Async context manager to create a redis connection
     """
 
+    redis = contextvars.ContextVar("redis-server")  # type: aioredis.Redis
+
     async def __aenter__(self):
-        self.conn = await aioredis.create_redis(os.environ["REDIS_URL"])
-        return self.conn
+        if AsyncRedis.redis.get(None) is None:
+            AsyncRedis.redis.set(
+                await aioredis.from_url(os.environ["REDIS_URL"], decode_responses=False)
+            )
+        return AsyncRedis.redis.get()
 
     async def __aexit__(self, exc_type, exc, tb):
-        self.conn.close()
-        await self.conn.wait_closed()
+        pass
```

Comment:
Given that this is no longer a context manager, could you try applying my suggestion from https://github.com/mozilla/libmozevent/pull/70#discussion_r703353880?